### PR TITLE
tests/unittests: fix GCC compiler bug in core-atomic for SAML1X

### PR DESCRIPTION
**Pull request by dylad**
### Contribution description

This PR provides a quick (dirty ?) fix for SAML1X families and unittests.
In fact, this PR extends a current fix that was there to fix a previous compiler issue when I was porting SAML10/SAML11 MCUs. I don't know if this is the best way to handle it...

### Testing procedure

Compile tests/unittests on SAML10 or SAML11 boards.

### Issues/PRs references
#13459

### CODE REVIEW

File: tests/unittests/tests-core/tests-core-atomic.c [hunk 1]
```diff
@@ -26,6 +26,11 @@ static void test_atomic_flag(void)
     TEST_ASSERT_EQUAL_INT(0, atomic_flag_test_and_set(&flag));
 }
 
+/* Prevent compiler optimization for SAML1X because of gcc internal bug */
+#ifdef CPU_SAML1X
```
Comment (benpicco) [inline tests/unittests/tests-core/tests-core-atomic.c]:
```suggestion
#ifdef CPU_ARCH_CORTEX_M23
```

**Comment by benpicco** [comment]
Murdock is happy, please squash

**Comment by benpicco** [comment]
You forgot to remove the saml1x boards from `BOARD_BLACKLIST`

**Comment by dylad** [comment]
> You forgot to remove the saml1x boards from BOARD_BLACKLIST

Which blacklist are you referring to ? I didn't find it.

**Comment by benpicco** [comment]
The one added by 13a6cfb9ba4547610d903ca4c356943a69a0a0a2

**Comment by dylad** [comment]
> The one added by 13a6cfb

Oh, this PR needs a rebase then... I thought I was blind or something.

**Comment by dylad** [comment]
Should be good now.

**Comment by benpicco** [comment]
You can squash directly :wink:

**Comment by dylad** [comment]
@kaspar030 any words before pushing the big button ?

File: boards/pba-d-01-kw2x/include/board.h [hunk 2]
```diff
@@ -85,6 +85,14 @@ extern "C"
 #define KW2XRF_SHARED_SPI          (0)
 /** @}*/
 
+/**
```
Comment (kaspar030) [inline boards/pba-d-01-kw2x/include/board.h]: unrelated?

**Comment by kaspar030** [comment]
> @kaspar030 any words before pushing the big button ?

Let's not forget to re-check on the next toolchain upgrade. Disabling atomics optimization cannot be good. :)

File: boards/pba-d-01-kw2x/include/board.h (hunk 2 above)
Comment (dylad) [reply boards/pba-d-01-kw2x/include/board.h]: Huh, yes ! I have no idea where does this come from...

**Comment by dylad** [comment]
> Let's not forget to re-check on the next toolchain upgrade. Disabling atomics optimization cannot be good. :)

I will try to remember it :) But seems like this bug is getting worse on newer toolchain

**Comment by kaspar030** [comment]
> But seems like this bug is getting worse on newer toolchain

really? did you try a gcc-10 release?

**Comment by dylad** [comment]
> really? did you try a gcc-10 release snapshot?

No but I ported this MCU one year ago and during the port I already had [this issue](https://github.com/RIOT-OS/RIOT/pull/10653#issuecomment-456013626).

At this time, only one function was affected but now there is three.

**Comment by kaspar030** [comment]
&go.

**Comment by dylad** [comment]
Thanks @kaspar030 !
I'll keep an eye on it with newer GCC version.
